packages/core/src/utils.ts

```diff
@@ -1096,6 +1113,29 @@ export function concatUrls(baseUrl: string | URL, path: string): string {
   return new URL(ensureNoLeadingSlash(path), ensureTrailingSlash(baseUrl.toString())).toString();
 }
 
+/**
+ * Concatenates a given base URL and path, ensuring the URL has the appropriate `ws://` or `wss://` protocol instead of `http://` or `https://`.
+ *
+ * @param baseUrl - The base URL.
+ * @param path - The URL to concat. Can be relative or absolute.
+ * @returns The concatenated WebSocket URL.
+ */
 export function getWebSocketUrl(baseUrl: URL | string, path: string): string {
   return concatUrls(baseUrl, path).toString().replace('http://', 'ws://').replace('https://', 'wss://');
 }
+
+/**
+ * Converts the given `query` to a string.
+ *
+ * @param query - The query to convert. The type can be any member of `QueryTypes`.
+ * @returns The query as a string.
+ */
+export function getQueryString(query: QueryTypes): string {
+  if (typeof query === 'object' && !Array.isArray(query) && !(query instanceof URLSearchParams)) {
+    query = Object.fromEntries(Object.entries(query).filter((entry) => entry[1] !== undefined));
```

Comment:
I'm trying to think if this could cause any behavioral differences

In the realm of URLs, a key without equals sign is different than key with equals sign and empty value.

i.e., `example.com?foo` is not the same as `example.com?foo=`

I don't know if there are any cases in FHIR where that distinction matters.... 🤔 

The FHIR spec specifically says " the parameters are a series of name=[value] pairs" (https://hl7.org/fhir/R4/search.html#Introduction) so this is probably ok.

Reply:
Not sure what you mean here, this is filtering out `undefined`s in the map that would cause otherwise unintended behavior, specifically when we had some cases of `{ hello: foo?.bar }` which would result in: `hello=undefined` in the query string

Reply:
But I see your point, its not really possible for a user to input `?foo=` with this function... maybe a feature and not a bug?